wetmap/src/components/reusables/plainTextInput/index.tsx

```diff
@@ -1,71 +1,71 @@
-import React, { useEffect, useRef, useState } from 'react';
-import style from './style.module.scss';
+import React, { InputHTMLAttributes, useRef, useState } from 'react';
 import Icon from '../../../icons/Icon';
+import ButtonIcon from '../buttonIcon';
 
-export type PlainTextInputProps = {
-  content:     string
-  readOnly:    boolean
-  placeHolder: string
-  onSubmit:    (value: string) => void
+import './style.scss';
+
+type TextInputProps = InputHTMLAttributes<HTMLInputElement>;
+type CustomInputProps = {
+  onSave: (value: string) => void
+  error?: any
+  value?: string
 };
 
-export default function PlainTextInput(props: PlainTextInputProps) {
+const PlainTextInput = React.forwardRef<HTMLInputElement, TextInputProps & CustomInputProps>(function PlainTextInput(props: TextInputProps & CustomInputProps, forwardedRef) {
   const [isEditModeOn, setIsEditModeOn] = useState(false);
-  const [inputValue, setInputValue] = useState(props.content);
+  const [value, setValue] = useState(props.value);
+  const ref = useRef<HTMLDivElement>(null);
 
-  const adjustTextareaHeight = (textarea: HTMLTextAreaElement | null) => {
-    if (textarea) {
-      textarea.style.height = textarea.scrollHeight + 'px';
+  const onKeyDown = (event: React.KeyboardEvent) => {
+    if (event.key === 'Enter') {
+      event.preventDefault();
     }
   };
 
-  const textRef = useRef<HTMLTextAreaElement>(null);
-  useEffect(() => {
-    adjustTextareaHeight(textRef.current);
-  }, []);
-
   return (
-    <div className={style.plainTextInput}>
+    <div className="ssrc-plain-text-input">
+      <input type="hidden" name={props.name} value={value} onChange={props.onChange} ref={forwardedRef} />
 
-      <textarea
-        placeholder={props.placeHolder}
-        ref={textRef}
-        name="body"
-        readOnly={props.readOnly || !isEditModeOn}
-        value={inputValue}
-        onChange={function (e) {
-          adjustTextareaHeight(e.target);
-          setInputValue(e.target.value);
+      <div
+        ref={ref}
+        onKeyDown={onKeyDown}
+        suppressContentEditableWarning={true}
+        className="ssrc-plain-text-input__textarea"
+        contentEditable={!props.readOnly && isEditModeOn}
+        onInput={function (e) {
+          setValue(e.currentTarget.innerHTML);
         }}
-      />
+      >
+        {props.value || (!isEditModeOn && props.placeholder)}
+      </div>
 
       {!props.readOnly && isEditModeOn && (
-        <div className={style.icon}>
-          <Icon
-            name="check-bold"
-            fill="green"
-            width="30px"
-            onClick={() => {
-              setIsEditModeOn(false);
-              props.onSubmit(inputValue);
-            }}
-          />
-        </div>
+        <ButtonIcon
+          className="btn-sm"
+          icon={<Icon name="check-bold" fill="green" />}
```

Comment:
can you refresh my memory, why is "fill" getting picked up by typescript here as an error?

Reply:
I've just fixed that in https://github.com/Freem11/divego/pull/28
The error happening because "fill" is not described in props type.
I've added all standard svg attributes as possible properties for the Icon component as all of them go to svg tag